### DISCUSSION TITLE
Custom operators

### DISCUSSION THREAD

**Discussion by urob** [JetBrains/ideavim#702] [Q&A] [answered]
Are custom operators supported or are there any plans to support them?

Here's a silly example that replaces `{motion}` with `x`s. Eg, sourcing the following code and pressing `gxiw` in vim will replace `word` with `xxxx`.

```vim
nnoremap gx :set opfunc=Redact<cr>g@
function! Redact(type)
    execute "normal `[v`]rx"
endfunction
```

But sourcing the same code in `ideavim` does not seem to work correctly. Specifically, after pressing `gx`, the status line does indicate `g@`, suggesting that we indeed entered operator pending mode. However, after entering a motion, `Redact()` does not seem to have any effect.

**Comment by urob** [comment]
Just for context: I'd like to use this to create an operator that executes `{motion}` in a Python Console (using `:action ExecuteInPyConsoleAction` in PyCharm). This would allow for things like `3grr` to run the next 3 lines of code (with appropriate mappings for `gr` and `grr`). 

Currently, the closest I can get to this is mapping `gr` to `<Action>(ExecuteInPyConsoleAction)`. This works but is more cumbersome, requiring something like `v{motion}gr<esc>` instead of a simple `gr{motion}`.

Another cool application would be an operator that applies `ReformatCode` to the provided `motion`.

**Reply by citizenmatt** [reply]
This is a nice idea, and I've just created a PR to add the `'operatorfunc'` option. See [JetBrains/ideavim#785](https://github.com/JetBrains/ideavim/pull/785).

**Reply by citizenmatt** [reply, marked as answer]
This has just been merged. It should be available in the next release (probably 2.10), and available in EAP builds before then.
